iocore/net/SSLNetVConnection.cc

```diff
@@ -181,6 +180,12 @@ make_ssl_connection(SSL_CTX *ctx, SSLNetVConnection *netvc)
       BIO *wbio = BIO_new_fd(netvc->get_socket(), BIO_NOCLOSE);
       BIO_set_mem_eof_return(wbio, -1);
       SSL_set_bio(ssl, rbio, wbio);
+
+#if TS_OPENSSL_1_1_1_API
+      if (netvc->early_data_buf == nullptr) {
+        netvc->early_data_buf = static_cast<uint8_t *>(ats_malloc(SSLConfigParams::server_max_early_data));
```

Comment:
So we will malloc() this even when there is no early data? That seems *really* suboptimal. Can't you defer to the allocation to the case where there actually is early data in the TLS handshake?

Reply:
Yeah, I forgot to change it after testing, changing it now.